pywbem_mock/_wbemconnection_mock.py

```diff
@@ -995,6 +1009,32 @@ def _display_objects(obj_type, object_repo, namespace, cmt_begin, cmt_end,
                     else:
                         _uprint(dest, obj.tomof())
 
+    def register_provider(self, namespace, classname, provider):
+        """
+        Register the provider object for namespace and class.
```

Comment:
Right now, we are only supporting user-provided instance providers. We have two options:
* Do not yet introduce a notion of provider types. In that case, this function should document that it registers instance providers (only).
* Introduce a notion of provider types already now, but support only instance providers at this point.

Reply:
These are all Instance providers, The remaining type is really the Method provider which is a significantly different bird I am afraid.  I will attack that tomorrow.

Reply:
right, all user provided providers are instance providers of some sort. Still, the C-U-D type, the method type, the association traversal type, the read-enum type are all different types. So when I said "instance provider" in my comment, I meant to say C-U-D instance provider. Anyway, it is just a comment about wording, so let's wait how it looks like when you have the method provider.

Reply:
I think I sorted that out in my mind late last night.  The provider registry should hiearchy of dictionaries where the top level is class name and within each classname are dict of namespaces, and dicts to represent each provider type if that provider exists for the defined class and namespaces.  Will try to create that this morning